dcos/subcommand.py

```diff
@@ -324,6 +405,95 @@ def _find_virtualenv(bin_directory):
     return virtualenv_path
 
 
+def _move(src, dst):
+    """Move file from src to the file or directory dst and remove dst if failure
+
+    :param src: source file
+    :type src: str
+    :param dst: destination file or directory
+    :type dst: str
+    :returns: None
+    """
+
+    try:
+        shutil.move(src, dst)
+    except Exception as e:
+        logger.exception(e)
+        shutil.rmtree(dst)
```

Comment:
I don't think this is enough, no? I think we want a `contextmanager` for `pkg_dir` defined in line 305. We want to delete the `pkg_dir` if we had any failure during the installation, no?